### PR TITLE
[16.0][MIG] maintenance_team_hierarchy: Migration to 16.0

### PR DESCRIPTION
**Pull request by kaynnan**
cc @marcelsavegnago 

#240

### CODE REVIEW

**Comment by kaynnan** [comment]
ping @dalonsod

File: maintenance_team_hierarchy/models/maintenance_team.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+class MaintenanceTeam(models.Model):
+
+    _parent_name = "parent_id"
+    _parent_store = True
+    _parent_order = "name"
+    _inherit = "maintenance.team"
+
+    parent_id = fields.Many2one("maintenance.team", ondelete="restrict")
+    parent_path = fields.Char(index=True)
```
Comment (dalonsod) [inline maintenance_team_hierarchy/models/maintenance_team.py]: As https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0 recommends, it should be useful adding this, not tested:
```suggestion
    parent_path = fields.Char(index=True, unaccent=False)
```

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only code review at this moment

File: maintenance_team_hierarchy/tests/test_team_hierarchy.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase, tagged
+
+
+# This test should only be executed after all modules have been installed.
+@tagged("post_install", "-at_install")
+class TestTeamHierarchy(TransactionCase):
+    def setUp(self):
```
Comment (dalonsod) [inline maintenance_team_hierarchy/tests/test_team_hierarchy.py]: And also replace `setUp` usage by `setUpClass`.

Comment (kaynnan) [reply maintenance_team_hierarchy/tests/test_team_hierarchy.py]: Thanks for the corrections, I made the adjustments

Comment (kaynnan) [reply maintenance_team_hierarchy/tests/test_team_hierarchy.py]: Thanks for the feedback, I made the adjustments regarding what was mentioned

**Review by dalonsod — APPROVED** [review]
LGTM

**Review by marcelsavegnago — APPROVED** [review]
LGTM

**Comment by kaynnan** [comment]
ping @etobella

**Review by etobella — APPROVED** [review]
LGTM 
/ocabot migration maintenance_team_hierarchy
/ocabot merge nobump

**Pull request merged**